doc/optimizers.md

```diff
@@ -501,6 +501,97 @@ optimizer.Optimize(f, coordinates);
+
+Note that the parameters `lowerBound` and `upperBound` are overloaded. Data types of `double` or `arma::mat` may be used. If they are initialized as single values of `double`, then the same value of the bound applies to all the axes, resulting in an initialization following a uniform distribution in a hypercube. If they are initialized as matrices of `arma::mat`, then the value of `lowerBound[i]` applies to axis `[i]`; similarly, for values in `upperBound`. This results in an initialization following a uniform distribution in a hyperrectangle within the specified bounds.
+
+Attributes of the optimizer may also be changed via the member methods
+`PopulationSize()`, `MaxGenerations()`, `CrossoverRate()`, `DistributionIndex()`, `Eta()`, `Epsilon()`, `LowerBound()` and `UpperBound()`.
+
+#### Examples
+
+<details open>
+<summary>Click to collapse/expand example code.
+</summary>
+
+```c++
+SchafferFunctionN1<arma::mat> SCH;
+arma::vec lowerBound("-1000");
+arma::vec upperBound("1000");
+AGEMOEA opt(50, 1000, 0.6, 20, 1e-6, 20, lowerBound, upperBound);
+
+typedef decltype(SCH.objectiveA) ObjectiveTypeA;
+typedef decltype(SCH.objectiveB) ObjectiveTypeB;
+
+arma::mat coords = SCH.GetInitialPoint();
+std::tuple<ObjectiveTypeA, ObjectiveTypeB> objectives = SCH.GetObjectives();
+
+// obj will contain the minimum sum of objectiveA and objectiveB found on the best front.
+double obj = opt.Optimize(objectives, coords);
+// Now obtain the best front.
+arma::cube bestFront = opt.ParetoFront();
+```
+
+</details>
+
+<details open>
+<summary>Click to collapse/expand example code.
+</summary>
+
+```c++
+ZDT3<> ZDT_THREE(300);
+const double lowerBound = 0;
+const double upperBound = 1;
+
+AGEMOEA opt(50, 500, 0.8, 20, 1e-6, 20, lowerBound, upperBound);
+typedef decltype(ZDT_THREE.objectiveF1) ObjectiveTypeA;
+typedef decltype(ZDT_THREE.objectiveF2) ObjectiveTypeB;
+bool success = true;
+arma::mat coords = ZDT_THREE.GetInitialPoint();
+std::tuple<ObjectiveTypeA, ObjectiveTypeB> objectives = ZDT_THREE.GetObjectives();
+opt.Optimize(objectives, coords);
+const arma::cube bestFront = opt.ParetoFront();
+  
+NSGA2 opt2(50, 5000, 0.5, 0.5, 1e-3, 1e-6, lowerBound, upperBound);
+// obj2 will contain the minimum sum of objectiveA and objectiveB found on the best front.
+double obj2 = opt2.Optimize(objectives, coords);
+ 
+arma::cube NSGAFront = opt2.ParetoFront();
+// Get the IGD score for NSGA front using AGEMOEA as reference.
+double igd =  IGD::Evaluate(NSGAFront, bestFront, 1);
+std::cout << igd << std::endl;
+```
+
+</details>
+
+#### See also:
+
+ * [An adaptive evolutionary algorithm based on non-euclidean geometry for many-objective optimization](https://doi.org/10.1145/3321707.3321839)
+ * [Multi-Objective Optimization in Wikipedia](https://en.wikipedia.org/wiki/Multi-objective_optimization)
+ * [Performance Indicators](#performance-indicators)
```

Comment:
Could be nice to also add this link to NSGA and MOEA documentation too---but up to you (and doesn't need to happen in this PR).